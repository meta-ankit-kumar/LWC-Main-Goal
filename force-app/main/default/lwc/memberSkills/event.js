export class MemberAddedEvent extends CustomEvent {
	static type = 'memberadded';
	constructor(value) {
		super(MemberAddedEvent.type, {
			detail: value
		})
	}
}