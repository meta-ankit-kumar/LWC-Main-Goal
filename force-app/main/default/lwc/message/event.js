export class ClearMessageEvent extends CustomEvent {
	static type='clearmessage';
	constructor(value) {
		super(ClearMessageEvent.type, { detail: { value }});
	}
}