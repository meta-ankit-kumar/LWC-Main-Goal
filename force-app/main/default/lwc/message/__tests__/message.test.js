import { createElement } from 'lwc';
import Message from 'c/message';
import { propsWithAutoClose, propsWithoutAutoClose } from './data';
import { ClearMessageEvent } from '../event';

// Helper function to create and set props for the component
function createMessage(props) {
    const element = createElement('c-member-info', {
        is: Message
    });

    Object.assign(element, props);

    document.body.appendChild(element);

    return element;
}

describe('c-message', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('should render the component successfully', () => {
        // GIVEN WHEN
        const element = createMessage(propsWithoutAutoClose);

        // THEN
        expect(element).toMatchSnapshot();
    });

    it('should emit ClearMessageEvent when autoClose is true', async () => {
        // GIVEN
        const element = createMessage(propsWithAutoClose);

        const eventListener = jest.fn();
        element.addEventListener(ClearMessageEvent.type, eventListener);

        // Act
        document.body.appendChild(element);

        const wait = () => {
            return new Promise(resolve => setTimeout(resolve, element.visibilityTime));
        };
        await wait();

        // THEN
        expect(eventListener).toHaveBeenCalled();
        expect(eventListener.mock.calls[0][0]).toBeInstanceOf(CustomEvent);
    });
});
