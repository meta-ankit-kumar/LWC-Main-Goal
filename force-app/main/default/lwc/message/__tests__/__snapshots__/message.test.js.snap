// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-message should render the component successfully 1`] = `
<c-member-info
  x-test_message-host=""
>
  #shadow-root(open)
    <div
      class="message-info"
      x-test_message=""
    >
      Test message
    </div>
</c-member-info>
`;
